packages/suite/src/components/suite/FormattedCryptoAmount/index.tsx

```diff
@@ -52,23 +57,23 @@ export const FormattedCryptoAmount = ({
 
     const areSatsSupported = !!networkFeatures?.includes('amount-unit');
 
-    let formattedValue = value;
+    let formattedValue = Number(value);
     let formattedSymbol = symbol?.toUpperCase();
 
     const isSatoshis = areSatsSupported && areSatsDisplayed;
 
     if (isSatoshis) {
-        formattedValue = formatCurrencyAmount(
-            Number(networkAmountToSatoshi(String(value), symbol as NetworkSymbol)),
-        ) as string;
+        formattedValue = Number(networkAmountToSatoshi(String(value), symbol as NetworkSymbol));
 
         formattedSymbol = isTestnet ? `sat ${symbol?.toUpperCase()}` : 'sat';
     }
 
     if (isRawString) {
         return (
             <>
-                {`${signValue ? `${isValuePositive(signValue) ? '+' : '-'}` : ''} ${formattedValue}
+                {`${
+                    signValue ? `${isValuePositive(signValue) ? '+' : '-'}` : ''
+                } ${formatCurrencyAmount(formattedValue, locale)}
                 ${formattedSymbol}`}
```

Comment:
A little hard to read, can we move some of the if condition out side the render function. e.g
```suggestion
               {displaySignValue} {formatCurrencyAmount(formattedValue, locale)} {formattedSymbol}
```

Reply:
3aa96a54055bc0b98fa0be83e3e90e6936d51552